From be47d6406a715aaba4664b1a01a64491e0b75056 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Fri, 10 Dec 2021 12:46:37 -0600
Subject: [PATCH 03/16] libfetch: use pkg-config to find openssl cflags

on mac, openssl is usually provided by Homebrew or some other third-party
package management system, which means pkg-config is needed to find it.

we already use pkg-config to find openssl when building apk itself.

ref #10794
---
 libfetch/Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libfetch/Makefile b/libfetch/Makefile
index c4b56f5..0c2617f 100644
--- a/libfetch/Makefile
+++ b/libfetch/Makefile
@@ -4,6 +4,13 @@ generate-y		+= ftperr.h httperr.h
 CFLAGS_common.o		+= -DCA_CERT_FILE=\"$(CONFDIR)/ca.pem\" -DCA_CRL_FILE=\"$(CONFDIR)/crl.pem\"
 CFLAGS_common.o		+= -DCLIENT_CERT_FILE=\"$(CONFDIR)/cert.pem\" -DCLIENT_KEY_FILE=\"$(CONFDIR)/cert.key\"
 
+PKG_CONFIG		?= pkg-config
+
+OPENSSL_CFLAGS		:= $(shell $(PKG_CONFIG) --cflags openssl)
+OPENSSL_LIBS		:= $(shell $(PKG_CONFIG) --libs openssl)
+
+CFLAGS_ALL		+= $(OPENSSL_CFLAGS)
+
 quiet_cmd_generr = GENERR  $@
       cmd_generr = $(src)/errlist.sh $(basename $(<F))_errlist $(shell echo $(basename $(<F)) | tr a-z A-Z) $< > $@
 
-- 
2.34.1

